Source/Core/Core/HW/WiimoteReal/IOLinux.cpp

```diff
@@ -137,30 +137,60 @@ bool WiimoteLinux::ConnectInternal()
   addr.l2_family = AF_BLUETOOTH;
   addr.l2_bdaddr = m_bdaddr;
   addr.l2_cid = 0;
+  int retry;
 
   // Output channel
   addr.l2_psm = htobs(WM_OUTPUT_CHANNEL);
-  if ((m_cmd_sock = socket(AF_BLUETOOTH, SOCK_SEQPACKET, BTPROTO_L2CAP)) == -1 ||
-      connect(m_cmd_sock, (sockaddr*)&addr, sizeof(addr)) < 0)
+  if (m_cmd_sock = socket(AF_BLUETOOTH, SOCK_SEQPACKET, BTPROTO_L2CAP))
+  {
+    retry = 0;
```

Comment:
Could you change this to `int retry = 0;` and remove the above declaration, to limit the variable scope?